specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,267 @@
+        (m_silentSaveAs ? L"Silent Save As Enabled" : L"Silent Save As Disabled"), L"Info",
+        MB_OK);
+    return true;
+}
+//! [ToggleSilent]
+
+//! [ProgrammaticSaveAs]
+// Call SaveContentAs method to trigger the programmatic save as.
+bool ScenarioSaveAs::ProgrammaticSaveAs()
+{
+    if (!m_webView2Staging20)
+        return false;
+    m_webView2Staging20->SaveContentAs(
+        Callback<ICoreWebView2StagingSaveContentAsCompletedHandler>(
+            [this](HRESULT errorCode, COREWEBVIEW2_SAVE_CONTENT_AS_RESULTS result) -> HRESULT
+            {
+                // Show SaveContentAs returned result, optional
+                // Skip message box when the result is COREWEBVIEW2_SAVE_AS_OPEN_SYSTEM_DIALOG (0),
+                // to prevent it deactivates save as modal dialog focus on the window
+                if (result > 0)
+                {
+                    MessageBox(
+                        m_appWindow->GetMainWindow(),
+                        (L"Save As " + saveAsResultString[result]).c_str(), L"Info", MB_OK);
+                }
+                return S_OK;
+            })
+            .Get());
+    return true;
+}
+//! [ProgrammaticSaveAs]
+```
+# API Details
+## Win32 C++
+```c++
+/// Specifies save as type selection options for `ICoreWebView2Staging20`,
+/// used in `SaveAsRequestedEventArgs`
+///
+/// When the source is a html page, allows to select `HTML_ONLY`,
+/// `SINGLE_FILE`, `COMPLETE`; when the source is a non-html,
+/// only allows to select `DEFAULT`; otherwise, will deny the download
+/// and return `COREWEBVIEW2_SAVE_AS_TYPE_NOT_SUPPORTED`.
+///
+/// The content type/format is a MIME type, indicated by the source
+/// server side and identified by the browser. It’s not related to the
+/// file’s type or extension. MIME type of `text/html`,
+/// `application/xhtml+xml` are considered as html page.
+[v1_enum] typedef enum COREWEBVIEW2_SAVE_AS_TYPE {
+  /// Default to save for a non-html content. If it is selected for a html
+  /// page, it’s same as HTML_ONLY option.
+  COREWEBVIEW2_SAVE_AS_TYPE_DEFAULT,
+  /// Save the page as html
+  COREWEBVIEW2_SAVE_AS_TYPE_HTML_ONLY,
+  /// Save the page as mhtml
+  COREWEBVIEW2_SAVE_AS_TYPE_SINGLE_FILE,
+  /// Save the page as html, plus, download the page related source files in
+  /// a folder
+  COREWEBVIEW2_SAVE_AS_TYPE_COMPLETE,
+} COREWEBVIEW2_SAVE_AS_TYPE;
+
+/// Status of a programmatic save as call, indicates the result
+/// for method `SaveContentAs`
+[v1_enum] typedef enum COREWEBVIEW2_SAVE_CONTENT_AS_RESULTS {
+  /// Programmatically open a system default save as dialog
+  COREWEBVIEW2_SAVE_AS_OPEN_SYSTEM_DIALOG,
+  /// Save as downloading not start as given an invalid path
+  COREWEBVIEW2_SAVE_AS_INVALID_PATH,
+  /// Save as downloading not start as given a duplicate filename and
+  /// replace file not allowed
+  COREWEBVIEW2_SAVE_AS_FILE_ALREADY_EXISTS,
+  /// Save as downloading not start as the `SAVE_AS_TYPE` selection not
```

Comment:
Can you elaborate on what this means? Can you give two examples one for not supported because of the content MIME type, and one because of system limits?